File: backend/app/models/table.py
from pydantic import BaseModel
from typing import Optional, Any
from decimal import Decimal
from datetime import datetime

class TableBase(BaseModel):
    """Base para mesa"""
    table_number: int
    is_occupied: bool = False
    x: int = 0
    y: int = 0

class TableCreate(TableBase):
    """Modelo para crear mesa"""
    pass

class TableUpdate(BaseModel):
    """Modelo para actualizar mesa"""
    table_number: Optional[int] = None
    is_occupied: Optional[bool] = None
    x: Optional[int] = None
    y: Optional[int] = None

class ActiveOrderInfo(BaseModel):
    id: int
    customer_name: Optional[str] = None
    total: Decimal
    created_at: datetime
    time_elapsed: str

class Table(TableBase):
    """Modelo completo de mesa con info de orden activa"""
    id: int
    active_order: Optional[ActiveOrderInfo] = None

    class Config:
        from_attributes = True

File: frontend/BurgerPOS/Models/Table.cs
using System.Text.Json.Serialization;

namespace BurgerPOS.Models;

public class Table
{
    [JsonPropertyName("id")]
    public int Id { get; set; }

    [JsonPropertyName("table_number")]
    public int TableNumber { get; set; }

    [JsonPropertyName("is_occupied")]
    public bool IsOccupied { get; set; }

    [JsonPropertyName("x")]
    public int X { get; set; }

    [JsonPropertyName("y")]
    public int Y { get; set; }

    [JsonPropertyName("active_order")]
    public ActiveOrderInfo? ActiveOrder { get; set; }
}

public class ActiveOrderInfo
{
    [JsonPropertyName("id")]
    public int Id { get; set; }

    [JsonPropertyName("customer_name")]
    public string? CustomerName { get; set; }

    [JsonPropertyName("total")]
    public decimal Total { get; set; }

    [JsonPropertyName("created_at")]
    public DateTime CreatedAt { get; set; }
    
    [JsonPropertyName("time_elapsed")]
    public string? TimeElapsed { get; set; }
}
